src/shared/components/query/QueryStore.ts

```diff
@@ -492,7 +493,9 @@ export class QueryStore
 		default: {},
 	});
 
-	readonly virtualStudies = remoteData(sessionServiceClient.getUserVirtualStudies(), []);
+	readonly virtualStudies = remoteData(async ()=>{
+		return sessionServiceClient.getUserVirtualStudies();

```

Comment:
I think this will still trigger an exception in the backend and http 500 error in the frontend if no session service is configured. As a side note, I also think that sessionService is an implementation detail that perhaps does not belong here. In this layer, all we want is the list of user virtual studies, perhaps from a VirtualStudiesService. Where the data is stored should be irrelevant here. VirtualStudiesService could then have the appropriate logic to return an empty list, e.g. if not session service is found, and do the exception handling/throwing if necessary.

Reply:
Hey Peter, you're right that we should't be calling the service at all if it's not configured. I will follow up with that.  Less sure on your second point.  The store is where we load data and handle the asynchronous behavior of doing so.   If there was a way of storing user virtual studies without session service, then we could further abstract, but it's the instance of remoteData that allows us to handle async, so this would more or less look like it does.   What I think is elegant about the remoteData (wrapper for mobx promise) paradigm is that it allows us to store both the data and the loading state of the data (and the potential error state) in one "node".